pkg/proxy/util/iptables/traffic.go

```diff
@@ -90,3 +90,64 @@ func (d *detectLocalByCIDR) JumpIfNotLocal(args []string, toChain string) []stri
 	klog.V(4).InfoS("Detect Local By CIDR", "cidr", d.cidr, "jumpNotLocal", line)
 	return line
 }
+
+type detectLocalByInterfaceName struct {
+	ifaceName string
+}
+
+// NewDetectLocalByInterfaceName implements the LocalTrafficDetector interface using an interface name.
+// This can be used when a pod interface name/prefix can be used to capture the notion of local traffic.
```

Comment:
maybe we can document this better, right now is misleading without Dan's context https://github.com/kubernetes/kubernetes/pull/95400#issuecomment-1039185009

Reply:
```suggestion
// NewDetectLocalByInterfaceName implements the LocalTrafficDetector interface using an interface name.
// This can be used when an interface name/prefix can be used to capture the notion of local traffic.
```

Reply:
I think that just removing pod ...

Reply:
ack, let me change this.

Reply:
done.